backend/app/verification.py

```diff
@@ -285,8 +307,25 @@ def _check_app_id(appid: str, login):
     if not utils.is_valid_app_id(appid):
         raise HTTPException(status_code=400, detail=ErrorDetail.MALFORMED_APP_ID)
 
-    if appid not in login["user"].dev_flatpaks(sqldb):
-        raise HTTPException(status_code=401, detail=ErrorDetail.NOT_APP_DEVELOPER)
+    if new_app:
+        if login["user"].accepted_publisher_agreement_at is None:
+            raise HTTPException(
+                status_code=403, detail=ErrorDetail.MUST_ACCEPT_PUBLISHER_AGREEMENT
+            )
+
+        if models.BackendApp.by_app_id(sqldb, appid) is not None:
+            raise HTTPException(status_code=400, detail=ErrorDetail.APP_ALREADY_EXISTS)
+
+        if _is_github_app(appid):
+            raise HTTPException(status_code=400, detail=ErrorDetail.APP_ALREADY_EXISTS)
+
+        if _matches_prefixes(appid, "com.github", "com.gitlab"):
```

Comment:
This means you can't migrate your grandfathered app, with those ids, unless you rebase onto a new id. Which might be fine.